### PR TITLE
Add basic support for the EVM version Paris

### PR DESCRIPTION
**Pull request by axic**
Depends on #13532.
Part of #13517.

### CODE REVIEW

**Comment by nikola-matic** [comment]
Since the dependency has been merged in, I've resolved the conflicts and rebased, so the PR should now be ready for review.

**Comment by axic** [comment]
Please note the last commit is there to show the test is indeed run.

File: Changelog.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ Language Features:
 Compiler Features:
  * Commandline Interface: Return exit code ``2`` on uncaught exceptions.
  * Commandline Interface: Add `--no-cbor-metadata` that skips CBOR metadata from getting appended at the end of the bytecode.
+ * EVM: Basic support for the EVM version "Paris".
```
Comment (cameel) [inline Changelog.md:9]: I see we did not add changelog for introducing some recent versions (London, Berlin, Istanbul), only for making them default. We used to do that with earlier versions though.

**Review by cameel — CHANGES_REQUESTED** [review]
Looks fine apart from the new version not being listed in the docs:
- List of EVM versions in "Using the Compiler".
- `evmVersion` option docs for StandardJSON.
- `--evm-version` option description on the CLI.

**Comment by cameel** [comment]
Maybe we should also update the description of `difficulty()` in Yul docs as a part of this PR?

**Comment by axic** [comment]
Updated everything.

> Maybe we should also update the description of difficulty() in Yul docs as a part of this PR?

Perhaps would leave this to the other PR when the changes are actually applied?

**Comment by cameel** [comment]
But I meant the changes that already took effect with the merge. I.e. the fact that it does not really return difficulty now (and that it depends on the underlying EVM rather than the EVM version set in the compiler).

**Review by cameel — DISMISSED** [review]

File: docs/yul.rst [hunk 2]
```diff
@@ -948,6 +948,11 @@ the ``dup`` and ``swap`` instructions as well as ``jump`` instructions, labels a
   You need to use the ``returndatasize`` opcode to check which part of this memory area contains the return data.
   The remaining bytes will retain their values as of before the call.
 
+.. note::
```
Comment (axic) [inline docs/yul.rst:951]: @cameel how about this?

File: docs/using-the-compiler.rst [hunk 3]
```diff
@@ -172,7 +172,8 @@ at each version. Backward compatibility is not guaranteed between each version.
      the optimizer.
 - ``london`` (**default**)
    - The block's base fee (`EIP-3198 <https://eips.ethereum.org/EIPS/eip-3198>`_ and `EIP-1559 <https://eips.ethereum.org/EIPS/eip-1559>`_) can be accessed via the global ``block.basefee`` or ``basefee()`` in inline assembly.
-
+- ``paris``
+   - No changes, however the semantics of the `difficulty` value have changed (see `EIP-4399 <https://eips.ethereum.org/EIPS/eip-4399>`_).
```
Comment (cameel) [inline docs/using-the-compiler.rst]:
```suggestion
   - No changes, however the semantics of the ``difficulty`` value have changed (see `EIP-4399 <https://eips.ethereum.org/EIPS/eip-4399>`_).
```

File: docs/yul.rst (hunk 2 above)
Comment (cameel) [reply docs/yul.rst:951]: Looks reasonable. I'd just add something like "see note below" in the table to draw attention to it.

**Review by cameel — DISMISSED** [review]
Looks good, now, apart from two minor things. I already fixed them and I'll push that to the PR in a moment.

File: docs/using-the-compiler.rst (hunk 3 above)
Comment (cameel) [reply docs/using-the-compiler.rst]: Fixed.

File: docs/yul.rst (hunk 2 above)
Comment (cameel) [reply docs/yul.rst:951]: Added.

**Comment by cameel** [comment]
After checking it out locally I spotted a few more places where the later EVM versions were missing. So I added them. That's already in the PR now. Some of those were in the fuzzer and tests so let's see if that doesn't break the CI.

**Review by cameel — DISMISSED** [review]

**Comment by axic** [comment]
@cameel yay, it is fixed now, let's merge :)

File: Changelog.md (hunk 1 above)
Comment (ekpyron) [reply Changelog.md:9]: We can adjust the changelog entry to ``Support for paris`` anyways, once we have the ``prevrandao`` change (which will be before the next release anyways).

**Review by ekpyron — APPROVED** [review]

**Review by cameel — APPROVED** [review]

**Pull request merged**